src/main/java/model/SongDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+package model;
+
+import org.sqlite.SQLiteDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class SongDaoImpl implements Dao<Song> {
```

Comment:
You combine an in memory repository and a sqlite db repository. Which of those will be the source of truth if there is a conflict between data in memory and data on disk? Another problem is that you don't check if your DAO already contains the song you want to add. I believe it would be reasonable to treat two songs with the same title, artist, album and year as equal, but to exclude id from comparing since it is purely a database-related field.